packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts

```diff
@@ -111,90 +144,117 @@ export class AuthService {
-      await this.logout(userEntity);
+    if (refreshTokenEntity) {
+      await this.tokenRepository.deleteOne(refreshTokenEntity);
     }
 
-    await this.authRepository.create({
-      user: userEntity,
-      refreshToken: refreshTokenHashed,
-      accessToken: accessTokenHashed,
-    });
+    const newRefreshTokenEntity = new TokenEntity();
+    newRefreshTokenEntity.user = userEntity;
+    newRefreshTokenEntity.type = TokenType.REFRESH;
+    const date = new Date();
+    newRefreshTokenEntity.expiresAt = new Date(
+      date.getTime() + this.refreshTokenExpiresIn,
+    );
+
+    await this.tokenRepository.createUnique(newRefreshTokenEntity);
 
-    return { accessToken, refreshToken };
+    return { accessToken, refreshToken: newRefreshTokenEntity.uuid };
   }
 
   public async forgotPassword(data: ForgotPasswordDto): Promise<void> {
-    const userEntity = await this.userService.getByEmail(data.email);
+    const userEntity = await this.userRepository.findByEmail(data.email);
 
     if (!userEntity) {
-      throw new NotFoundException(ErrorUser.NotFound);
+      throw new AuthError(ErrorUser.NotFound);
     }
 
-    if (userEntity.status !== UserStatus.ACTIVE)
-      throw new UnauthorizedException(ErrorAuth.UserNotActive);
+    if (userEntity.status !== UserStatus.ACTIVE) {
+      throw new AuthError(ErrorUser.UserNotActive);
+    }
 
-    const existingToken = await this.tokenRepository.findOne({
-      userId: userEntity.id,
-      tokenType: TokenType.PASSWORD,
-    });
+    const existingToken = await this.tokenRepository.findOneByUserIdAndType(
+      userEntity.id,
+      TokenType.PASSWORD,
+    );
 
     if (existingToken) {
-      await existingToken.remove();
+      await this.tokenRepository.deleteOne(existingToken);
     }
 
-    const newTokenEntity = await this.tokenRepository.create({
-      tokenType: TokenType.PASSWORD,
-      user: userEntity,
-    });
+    const tokenEntity = new TokenEntity();
+    tokenEntity.type = TokenType.PASSWORD;
+    tokenEntity.user = userEntity;
+    const date = new Date();
+    tokenEntity.expiresAt = new Date(
+      date.getTime() + this.forgotPasswordTokenExpiresIn,
+    );
+
+    await this.tokenRepository.createUnique(tokenEntity);
 
     await this.sendgridService.sendEmail({
       personalizations: [
         {
           to: data.email,
           dynamicTemplateData: {
             service_name: SERVICE_NAME,
-            url: `${this.feURL}/reset-password?token=${newTokenEntity.uuid}`,
+            url: `${this.feURL}/reset-password?token=${tokenEntity.uuid}`,
           },
         },
       ],
       templateId: SENDGRID_TEMPLATES.resetPassword,
     });
   }
 
-  public async restorePassword(data: RestorePasswordDto): Promise<boolean> {
-    const tokenEntity = await this.tokenRepository.findOne({
-      uuid: data.token,
-      tokenType: TokenType.PASSWORD,
-    });
+  public async restorePassword(
+    data: RestorePasswordDto,
+    ip?: string,
+  ): Promise<void> {
```

Comment:
add hcaptcha token verification even though you leave it commented

Reply:
added and commented